File: app/src/main/java/datanapps/roomapi/roomdatabase/BookRepository.java
package datanapps.roomapi.roomdatabase;

import android.content.Context;
import android.os.AsyncTask;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Room;

public class BookRepository {

    private String DB_NAME = "db_task";
    private AppDB noteDatabase;


    public BookRepository(Context context) {
        noteDatabase = Room.databaseBuilder(context, AppDB.class, DB_NAME).build();
    }

    public void insertTask(final Book note) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                noteDatabase.daoAccess().insertTask(note);
                return null;
            }
        }.execute();
    }

    public void updateTask(final Book note) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                noteDatabase.daoAccess().updateTask(note);
                return null;
            }
        }.execute();
    }

    public void deleteTask(final int id) {
        final LiveData<Book> task = getTask(id);
        if(task != null) {
            new AsyncTask<Void, Void, Void>() {
                @Override
                protected Void doInBackground(Void... voids) {
                    noteDatabase.daoAccess().deleteTask(task.getValue());
                    return null;
                }
            }.execute();
        }
    }

    public void deleteTask(final Book note) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                noteDatabase.daoAccess().deleteTask(note);
                return null;
            }
        }.execute();
    }

    public LiveData<Book> getTask(int id) {
        return noteDatabase.daoAccess().getTask(id);
    }

    public LiveData<List<Book>> getTasks() {
        return noteDatabase.daoAccess().fetchAllTasks();
    }

}

File: app/src/main/java/datanapps/roomapi/roomdatabase/BookRepository.kt
package datanapps.roomapi.roomdatabase

import android.content.Context

import androidx.lifecycle.LiveData
import androidx.room.Room
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class BookRepository(context: Context) {

    private val DB_NAME = "db_task"
    private val appDatabase: AppDB

    init {
        appDatabase = Room.databaseBuilder(context, AppDB::class.java, DB_NAME).build()
    }

    fun insertTask(note: Book) {

        GlobalScope.launch {
            appDatabase.daoBookAccess().insertTask(note)
        }
    }



    fun updateBook(book: Book) {
        GlobalScope.launch {
            appDatabase.daoBookAccess().updateTask(book)
        }
    }


    fun deleteBookById(id: Int) {
        val book = getBookById(id)
        if (book != null) {
            deleteBook(book.value!!)
        }
    }

    fun deleteBook(note: Book) {
        GlobalScope.launch {
            appDatabase.daoBookAccess().deleteTask(note)
        }
    }

    fun getBookById(id: Int): LiveData<Book>? {
        return appDatabase.daoBookAccess().getTask(id)
    }


    val getAllBookList: LiveData<List<Book>>
        get() = appDatabase.daoBookAccess().fetchAllTasks()

}
